src/binary_parser.rs

```diff
@@ -26,10 +26,20 @@ impl BinaryInfo {
 }
 
 /// Uses goblin to parse a binary file, returns information on symbols/bss/adjusted offset etc
-pub fn parse_binary(_pid: remoteprocess::Pid, filename: &str, addr: u64, size: u64) -> Result<BinaryInfo, Error> {
+pub fn parse_binary(_pid: remoteprocess::Pid, filename: &str, addr: u64, size: u64, _is_bin: bool) -> Result<BinaryInfo, Error> {
     // on linux the process could be running in docker, access the filename through procfs
+    // if filename is the binary executable (not libpython) - take it from /proc/pid/exe, which works
+    // across namespaces just like /proc/pid/root, and also if the file was deleted.
     #[cfg(target_os="linux")]
-    let filename = &format!("/proc/{}/root{}", _pid, filename);
+    let _tmp;
+    #[cfg(target_os="linux")]
+    let filename = if _is_bin {
```

Comment:
I don't like this part, too much boilerplate, but I'm not a Rust expert so I'm not sure how to fight the lifetime checker here... Would love a suggestion on how to improve this bit. (without the `_tmp`, I get the `creates a temporary which is freed while still in use` error)